From 5ca39688aa14fca3a9190140b3a4ee3b1c826341 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=8D=C3=B1igo=20Huguet?= <ihuguet@redhat.com>
Date: Fri, 21 Jan 2022 08:49:01 +0100
Subject: [PATCH 04/36] rtw89: fix return value check in
 rtw89_cam_send_sec_key_cmd()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Bugzilla: http://bugzilla.redhat.com/2033291

commit a04310edcd00d6014126483a2d8cd95b4786db25
Author: Yang Yingliang <yangyingliang@huawei.com>
Date:   Mon Oct 18 11:31:02 2021 +0800

    rtw89: fix return value check in rtw89_cam_send_sec_key_cmd()

    Fix the return value check which testing the wrong variable
    in rtw89_cam_send_sec_key_cmd().

    Reported-by: Hulk Robot <hulkci@huawei.com>
    Fixes: e3ec7017f6a2 ("rtw89: add Realtek 802.11ax driver")
    Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
    Acked-by: Ping-Ke Shih <pkshih@realtek.com>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
    Link: https://lore.kernel.org/r/20211018033102.1813058-1-yangyingliang@huawei.com

Signed-off-by: Íñigo Huguet <ihuguet@redhat.com>
---
 drivers/net/wireless/realtek/rtw89/cam.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtw89/cam.c b/drivers/net/wireless/realtek/rtw89/cam.c
index c1e8c76c6aca..ad7a8155dbed 100644
--- a/drivers/net/wireless/realtek/rtw89/cam.c
+++ b/drivers/net/wireless/realtek/rtw89/cam.c
@@ -77,7 +77,7 @@ static int rtw89_cam_send_sec_key_cmd(struct rtw89_dev *rtwdev,
 		return 0;
 
 	ext_skb = rtw89_cam_get_sec_key_cmd(rtwdev, sec_cam, true);
-	if (!skb) {
+	if (!ext_skb) {
 		rtw89_err(rtwdev, "failed to get ext sec key command\n");
 		return -ENOMEM;
 	}
-- 
2.13.6

